### DISCUSSION TITLE
How to execute/debug code?

### DISCUSSION THREAD

**Discussion by bigdatadeveloper8** [zed-industries/zed#6562] [Feature Requests]
I am newbie with zed editor, I am writing code in python,Go and rust though I really don't how to run the code and check the output. 

Please guide.

**Comment by dupeiran001** [comment]
seems terminal is the only way. cargo is great!!.

besides, debugging without GUI is quite inconvenient.

**Comment by JosephTLyons** [comment]
Running and debugging code from Zed isn't currently possible yet, here are some relevant issues you can upvote (👍):

- https://github.com/zed-industries/community/issues/811
- https://github.com/zed-industries/community/issues/455
- https://github.com/zed-industries/community/issues/668

**Comment by sheikalthaf** [comment]
@JosephTLyons  without this feature zed literally unusable at the movement, Also I see there is no plan yet to integrate this feature. I really love the ZED editor but need to wait until it gets integrated

**Reply by hovsater** [reply]
Zed is very much usable without having builtin debugging/testing capabilities. The integrated terminal allows you to run any command you might have in mind and the feature `editor: copy relative path` can make it easier to run tools on the current file.

Whenever there's news to share, we'll make sure to let you all know. 🙂

**Reply by sheikalthaf** [reply]
@hovsater yes, you are right. I really love the zed editor to quick open any files and find something. But as my daily driver debugging is more important. Command line not helping much in my case if I want to debug some code line by line.

I'm eagerly waiting for this feature to be integrate quickly

**Reply by DiegoAz** [reply]
@hovsater can I use the feature `editor: copy relative path` in some way to do something like the following, but instead of `utils.py` put some variable like $file ([Sublime Text style](https://www.sublimetext.com/docs/build_systems.html#variables))?
 ```json
"terminal": {
    "shell": {
      "with_arguments": {
        "program": ".venv/bin/ipython",
        "args": [
          "-i",
          "utils.py"
        ]
      }
    }
  }
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/support
